api/v1beta1/types.go

```diff
@@ -698,3 +698,29 @@ type AzureBastion struct {
 func IsTerminalProvisioningState(state ProvisioningState) bool {
 	return state == Failed || state == Succeeded
 }
+
+// KubeletConfig kubelet configurations of agent nodes.
+type KubeletConfig struct {
```

Comment:
this looks like it's a duplicate of the type in exp/? If not, where is it used?

Reply:
Rather than define our own `KubeletConfig` type spec, can we simply add a new property to the `AgentPoolSpec`:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/v1.0.1/azure/types.go#L435

Something like this:

```
$ git diff
diff --git a/azure/types.go b/azure/types.go
index 45bb67f1..e0422643 100644
--- a/azure/types.go
+++ b/azure/types.go
@@ -19,6 +19,7 @@ package azure
 import (
        "reflect"
 
+       "github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2021-05-01/containerservice"
        "github.com/google/go-cmp/cmp"
 
        infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
@@ -441,4 +442,7 @@ type AgentPoolSpec struct {
 
        // AvailabilityZones represents the Availability zones for nodes in the AgentPool.
        AvailabilityZones []string
+
+       // KubeletConfig kubelet configurations of agent nodes.
+       KubeletConfig containerservice.KubeletConfig
 }
```

Reply:
discussed this with @jackfrancis offline, we don't actually want to do this, because it would make us rely on the SDK not making breaking changes and prevents us from using conversion for back compat (and also ties our API to a specific SDK version)